src/Datadog.Trace/HttpOverStreams/DatadogHttpClient.cs

```diff
@@ -62,23 +63,22 @@ private HttpResponse ReadResponse(Stream responseStream)
             var stringBuilder = new StringBuilder();
 
             var chArray = new byte[1];
-            void GoNextChar()
+
+            async Task GoNextChar()
             {
-                streamPosition++;
-                chArray[0] = (byte)responseStream.ReadByte();
+                await responseStream.ReadAsync(chArray, offset: 0, count: 1);
                 currentChar = Encoding.ASCII.GetChars(chArray)[0];
+                streamPosition++;
             }
 
-            bool IsNewLine()
+            async Task<bool> IsNewLine()
             {
                 if (currentChar.Equals(DatadogHttpValues.CarriageReturn))
                 {
                     // end of headers
-                    if (DatadogHttpValues.CrLfLength > 1)
-                    {
-                        // Skip the newline indicator
-                        GoNextChar();
-                    }
+                    // Next character should be a LineFeed, regardless of Linux/Windows
+                    // Skip the newline indicator
```

Comment:
Should we confirm that the next byte is actually a line feed? Is there a risk that a carriage returns appears in the stream without a line feed following it?

Reply:
This is the same behaviour as we were using on Windows previously, A CR in the header without a LF would be incorrect according to the HTTP spec AFACIT. I guess we should throw if we encounter that?

Reply:
Updated in https://github.com/DataDog/dd-trace-dotnet/pull/1218/commits/bce14fb4452c93bc6866947d68dd25c5402e22b5 to throw if we don't get a LF